### PR TITLE
Create "Migrate from Log4j 2" page

### PR DESCRIPTION
**Pull request by ppkarwasz**
We create a "Migrate from Log4j 2" that specifies:

- the required changes in the runtime classpath due to the splitting of modules,
- a warning about the change of the naming convention for configuration properties,
- a list of migration rules for users that love the Java properties configuration format.

Part of #2540.

### CODE REVIEW

File: src/site/antora/modules/ROOT/pages/manual/migration.adoc [hunk 1]
```diff
@@ -16,5 +16,247 @@ Licensed to the Apache Software Foundation (ASF) under one or more
...
+| {sonatype-url}/log4j-config-properties[`log4j-config-properties`]
+
+| xref:manual/configuration.adoc#configuration-with-yaml[Configuration file: YAML format]
+| {sonatype-url}/log4j-config-yaml[`log4j-config-yaml`]
+
+| xref:manual/layouts.adoc#CSVLayouts[CSV layouts]
+| {sonatype-url}/log4j-csv[`log4j-csv`]
+
+| {log4j2-url}/manual/layouts.html#GELFLayout[GELF layout]
+| _removed without replacement_ footnote:removal[]
```
Comment (vy) [inline src/site/antora/modules/ROOT/pages/manual/migration.adoc:79]: Users can use JTL with `GelfLayout.json` event template.

File: src/site/antora/modules/ROOT/pages/manual/migration.adoc [hunk 2]
```diff
@@ -16,5 +16,247 @@ Licensed to the Apache Software Foundation (ASF) under one or more
...
+
+| xref:manual/appenders.adoc#JDBCPoolingDriver[JDBC appender: DBCP 2 connection source]
+| {sonatype-url}/log4j-jdbc-dbcp2[`log4j-jdbc-dbcp2`]
+
+| xref:manual/appenders.adoc#JDBCDataSource[JDBC appender: JNDI connection source]
+| {sonatype-url}/log4j-jndi[`log4j-jndi`]
+
+| {log4j2-url}/manual/appenders.html#JeroMQAppender[JeroMQ appender]
+| external
+https://github.com/fbacchella/loghublog4j2#zmqappender[`loghublog4j2` maintained by Fabrice Bacchella]
```
Comment (vy) [inline src/site/antora/modules/ROOT/pages/manual/migration.adoc:92]: I'd remove the `maintained by ...` part. It is not future-proof (maintainers come and go in time) , bloats the message, doesn't add value.

File: src/site/antora/modules/ROOT/pages/manual/migration.adoc [hunk 3]
```diff
@@ -16,5 +16,247 @@ Licensed to the Apache Software Foundation (ASF) under one or more
...
+
+| {log4j2-url}/log4j-taglib.html[JSP Tag library]
+| _removed without replacement_ footnote:removal[]
+
+| {log4j2-url}/manual/appenders.html#KafkaAppender[Kafka appender]
+| _removed without replacement_ footnote:removal[]
+
+| xref:manual/lookups.adoc#KubernetesLookup[Kubernetes lookup]
+| external
+https://github.com/fabric8io/kubernetes-client/blob/main/doc/KubernetesLog4j.md[`kubernetes-log4j` maintained by Fabric8]
```
Comment (vy) [inline src/site/antora/modules/ROOT/pages/manual/migration.adoc:111]: I'd remove the `maintained by ...` part. It is not future-proof (maintainers come and go in time) , bloats the message, doesn't add value.

File: src/site/antora/modules/ROOT/pages/manual/migration.adoc [hunk 4]
```diff
@@ -16,5 +16,247 @@ Licensed to the Apache Software Foundation (ASF) under one or more
...
+
+| {log4j2-url}/manual/webapp.html[Servlet application support: Java EE version]
+| _removed without replacement_ footnote:removal[]
+
+| {log4j2-url}/manual/appenders.html#SMTPAppender[SMTP appender: Java EE version]
+| _removed without replacement_ footnote:removal[]
+
+| {log4j2-url}/manual/lookups.html#_spring_boot_lookup[Spring Boot lookup]
+| integrated into
+https://central.sonatype.com/artifact/org.springframework.boot/spring-boot[`spring-boot`]
```
Comment (vy) [inline src/site/antora/modules/ROOT/pages/manual/migration.adoc:124]: It is not really integrated into the `spring-boot` artifact, right? I'd rather use the following:
```
Integrated to https://docs.spring.io/spring-boot/reference/features/logging.html#features.logging.log4j2-extensions.environment-properties-lookup[Spring Boot]
```

File: src/site/antora/modules/ROOT/pages/manual/migration.adoc [hunk 5]
```diff
@@ -16,5 +16,247 @@ Licensed to the Apache Software Foundation (ASF) under one or more
...
+To help you with the migration process a backward compatibility system has been implemented, which translates Log4j 2 property names to their Log4j 3 equivalents.
+
+If you use configuration properties to finely tune Log4j Core, make sure that:
+
+* either you use the naming convention from the
+xref:manual/systemproperties.adoc[Log4j 3 configuration properties]
+page (recommended),
+* or you use the naming convention from the
+{log4j2-url}/manual/systemproperties.html[Log4j 2 configuration properties]
+page.
```
Comment (vy) [inline src/site/antora/modules/ROOT/pages/manual/migration.adoc:142]: I'd rephrase the main message being conveyed here as follows:

* You are advised to migrate to the new naming convention.
* In the meantime, we covered your back with a name translator, granted you're following `insert-link-here:foo[the canonical/correct/etc. Log4j 2 naming convention]`

File: src/site/antora/modules/ROOT/pages/manual/migration.adoc [hunk 6]
```diff
@@ -16,5 +16,247 @@ Licensed to the Apache Software Foundation (ASF) under one or more
...
+appender.0.name = CONSOLE
+appender.1.type = File
+appender.1.name = MAIN
+appender.1.fileName = logs/main.log
+appender.2.type = File
+appender.2.name = DEBUG_LOG
+appender.2.fileName = logs/debug.log
+----
++
+you can write:
```
Comment (vy) [inline src/site/antora/modules/ROOT/pages/manual/migration.adoc:173]: `can` or `must`?

File: src/site/antora/modules/ROOT/pages/manual/migration.adoc (hunk 1 above)
Comment (ppkarwasz) [reply src/site/antora/modules/ROOT/pages/manual/migration.adoc:79]: Fixed in dbe9112e31ea5e41a910b408936b79dc507c8c3f

File: src/site/antora/modules/ROOT/pages/manual/migration.adoc (hunk 3 above)
Comment (ppkarwasz) [reply src/site/antora/modules/ROOT/pages/manual/migration.adoc:111]: Fixed in dbe9112e31ea5e41a910b408936b79dc507c8c3f

File: src/site/antora/modules/ROOT/pages/manual/migration.adoc (hunk 4 above)
Comment (ppkarwasz) [reply src/site/antora/modules/ROOT/pages/manual/migration.adoc:124]: The logging abstraction is actually part of `spring-boot`, but your version sounds better.

Fixed in dbe9112e31ea5e41a910b408936b79dc507c8c3f

File: src/site/antora/modules/ROOT/pages/manual/migration.adoc (hunk 2 above)
Comment (ppkarwasz) [reply src/site/antora/modules/ROOT/pages/manual/migration.adoc:92]: Fixed in dbe9112e31ea5e41a910b408936b79dc507c8c3f

File: src/site/antora/modules/ROOT/pages/manual/migration.adoc (hunk 5 above)
Comment (ppkarwasz) [reply src/site/antora/modules/ROOT/pages/manual/migration.adoc:142]: Fixed in dbe9112e31ea5e41a910b408936b79dc507c8c3f

File: src/site/antora/modules/ROOT/pages/manual/migration.adoc (hunk 6 above)
Comment (ppkarwasz) [reply src/site/antora/modules/ROOT/pages/manual/migration.adoc:173]: Its optional, although I would still use it in the examples, since it is much shorter (vertically).
